File: backend/PetCare.Domain/Billing/Payment.cs
using PetCare.Domain.Bookings;

namespace PetCare.Domain.Billing;

public class Payment
{
    public Guid Id { get; set; }
    public Guid BookingId { get; set; }
    public Booking Booking { get; set; } = null!;
    
    public decimal Amount { get; set; }
    public DateTime PaymentDate { get; set; }
    public string Status { get; set; } = "Pending"; // Pending, Completed, Failed
    public string PaymentMethod { get; set; } = string.Empty; // Online, OnLocation
}

File: frontend/lib/models/payment.dart
class Payment {
  final String id;
  final String bookingId;
  final double amount;
  final DateTime paymentDate;
  final String status;
  final String paymentMethod;

  Payment({
    required this.id,
    required this.bookingId,
    required this.amount,
    required this.paymentDate,
    required this.status,
    required this.paymentMethod,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: json['id'] ?? '',
      bookingId: json['bookingId'] ?? '',
      amount: (json['amount'] ?? 0).toDouble(),
      paymentDate: DateTime.parse(json['paymentDate']),
      status: json['status'] ?? 'Pending',
      paymentMethod: json['paymentMethod'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'bookingId': bookingId,
      'amount': amount,
      'paymentDate': paymentDate.toIso860String(),
      'status': status,
      'paymentMethod': paymentMethod,
    };
  }
}
